includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -451,6 +451,25 @@ public function cancel_subscription( WC_Subscription $subscription ) {
 		}
 	}
 
+	/**
+	 * Suspends the WCPay subscription when a WC subscription is put on-hold.
+	 *
+	 * @param WC_Subscription $subscription The WC subscription that was suspended.
+	 *
+	 * @return void
+	 */
+	public function handle_subscription_status_on_hold( WC_Subscription $subscription ) {
+		$this->suspend_subscription( $subscription );
```

Comment:
I've added logging that assumes the subscription is a WCPay Subscription. However, this might not be the case, e.g. a store that has WC Subscriptions active (currently, or in the past).

Reviewer question: should we check that the subscription is a WCPay Subscription before suspending, either here or in `suspend_subscription`?

Reply:
Update: I added this as an early-exit precondition. When I was testing I `Suspend`ed a regular tokenised subscription, and the MIA WCPay ID broke the logging.

So that's a behaviour change, but maybe a good thing.

Reply:
The way this worked previously was the `suspend_subscription()` function would call `update_subscription()` and that function would check if the subscription had a wcpay subscription ID. 

https://github.com/Automattic/woocommerce-payments/blob/2c22e38c0997c716909d7a9a367ee1f3363ed488/includes/subscriptions/class-wc-payments-subscription-service.php#L834-L841

This would log entries into the WC Payments logs that could trip people up thinking there was an issue or error, when in fact the subscription just wasn't a WCPay Subscription and shouldn't have gotten to that point. 

So, exiting early when the subscription isn't a wcpay subscription is a good idea. 👍 

---- 

By introducing a new function that is hooked onto `woocommerce_subscription_status_on-hold`, we'll also need to update these instances where we unhook these callbacks where we don't intend to notify Stripe of a suspension. 

This is the only time we currently do this: 
https://github.com/Automattic/woocommerce-payments/blob/68856b78e0b41f4c4c4af66b0182f656d06dc138/includes/subscriptions/class-wc-payments-subscriptions-event-handler.php#L127-L129

Reply:
Rather than have to do all this updating though, we could update `suspend_subscription()` instead and just have it exit early and add the new order notes and logs.

Reply:
Thanks – that makes sense. I didn't realise the implications of the new callback; the idea was that the logging happens outside `suspend_subscription`, nearer the "action" that triggered suspension, rather than generically inside `suspend`.

However, if we add a call stack, then the generic option would give us the same info (and more), and would be lower impact on the code base too.

Reply:
Just realised we won't be able to add a specific order note for status change (on-hold => suspend). Maybe that's ok, I was hoping to add individual notes for the two scenarios.

Reply:
Went with the decoupled status handler, so we get the actions logged and shown in notes: e5dce3f8a4c7ad8042052af2b8df38d69fb230d1

@james-allan  let me know what you think, happy to go with generic approach if that's better overall.

Reply:
This works for me.